In [4]:
import pandas as pd
df = pd.read_csv("tested.csv")

In [17]:
# information on the dataset
print(df.shape)
df.info()

(418, 12)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  418 non-null    int64  
 1   Survived     418 non-null    int64  
 2   Pclass       418 non-null    int64  
 3   Name         418 non-null    object 
 4   Sex          418 non-null    object 
 5   Age          418 non-null    float64
 6   SibSp        418 non-null    int64  
 7   Parch        418 non-null    int64  
 8   Ticket       418 non-null    object 
 9   Fare         417 non-null    float64
 10  Cabin        91 non-null     object 
 11  Embarked     418 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 39.3+ KB


In [8]:
df.isnull().sum()

,0
PassengerId,0
Survived,0
Pclass,0
Name,0
Sex,0
Age,86
SibSp,0
Parch,0
Ticket,0
Fare,1


 تم تعويض القيم المفقودة بمتوسط الحسابي للاعمار و ما تم الحذف لانه القيم المفقودة كثيرة و الداتا قليلة


In [16]:
#التخلص من القيم المفقودة
df["Age"].fillna(df["Age"].mean(),inplace = True)
df["Age"].isnull().sum()

np.int64(0)

تم حذف العمود بسبب اغلب القيم الموجودة فيه قيم مفقودة بحيث حجم الداتا 418 و القيم المفقودة 327

In [20]:
# drop Cabin
df=df.drop(columns=["Cabin"])
print(df.columns)

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Embarked'],
      dtype='object')


تقريبا نصف الاشخاص الموجودين على السفينه هم رجال

In [27]:
#number of children
child = df[df["Age"]<18]
print(child.shape)

male = df[df["Sex"] == "male"]
print (male.shape)

(41, 11)
(266, 11)


In [32]:
# نسبة النجاه للنساء اكبر بكثير من الرجال
print(df.groupby("Sex")["Survived"].mean())

# نسبة النجاة الدرجة الاولى تقريبا النصف
print(df.groupby("Pclass")["Survived"].mean())

Sex
female    1.0
male      0.0
Name: Survived, dtype: float64
Pclass
1    0.467290
2    0.322581
3    0.330275
Name: Survived, dtype: float64
